### PR TITLE
simple-dlna-browser: Add new package

### PR DESCRIPTION
**Pull request by ghost**
###### Description of changes

Add a  useful CLI script I found that's way light than VLC and other GUI apps since SSDP is pretty much a text-based protocol built on top of HTTP. It has no license attached so I marked it as unfree.
If the author adds a license, we will be able to specify it. @javier-lopez, would you be willing to add a license to your `simple-dlna-browser` script for its inclusion in NixOS?

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by javier-lopez** [comment]
Hello @LoveIsGrief , sure, would it be enough to add a header at the top of the script?

**Comment by javier-lopez** [comment]
I've licenced the script with the https://en.wikipedia.org/wiki/Fair_License , I hope this makes clear and easy the inclusion of the file, happy new year 🍻 , https://github.com/javier-lopez/learn/blob/master/sh/tools/simple-dlna-browser#L9

**Comment by LoveIsGrief** [comment]
Thank you @javier-lopez !

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -8352,6 +8352,13 @@
     githubId = 265084;
     name = "Jason O'Conal";
   };
+  loveisgrief = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:8355]: Please put this into a separate commit with the message ``maintainers: add NAME`` and sort it alphabetically.

File: pkgs/tools/networking/simple-dlna-browser/default.nix [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  buildInputs = with pkgs; [
+    gnused
+    socat
+  ];
+  binPaths = builtins.map (x: "${x}/bin") buildInputs;
+  paths = builtins.concatStringsSep ":" binPaths;
+in
+pkgs.stdenv.mkDerivation {
+  name = "simple-dlna-browser";
+  meta = {
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/simple-dlna-browser/default.nix:12]: meta must be last

File: pkgs/tools/networking/simple-dlna-browser/default.nix [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+let
+  buildInputs = with pkgs; [
+    gnused
+    socat
+  ];
+  binPaths = builtins.map (x: "${x}/bin") buildInputs;
+  paths = builtins.concatStringsSep ":" binPaths;
+in
+pkgs.stdenv.mkDerivation {
+  name = "simple-dlna-browser";
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/simple-dlna-browser/default.nix:11]: Please use pname+version

File: pkgs/tools/networking/simple-dlna-browser/default.nix [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+      [DLNA]: https://en.wikipedia.org/wiki/Digital_Living_Network_Alliance
+      [Kodi]: https://kodi.tv/
+      [VLC]: https://www.videolan.org/vlc/
+    '';
+    homepage = "https://github.com/javier-lopez/learn/blob/master/sh/tools/simple-dlna-browser";
+    license = lib.licenses.fair;
+    maintainers = with lib.maintainers; [ loveisgrief ];
+  };
+
+  src = pkgs.fetchgit {
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/simple-dlna-browser/default.nix:27]: fetchFromGitHub must be used

File: pkgs/tools/networking/simple-dlna-browser/default.nix [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    url = "https://github.com/javier-lopez/learn.git";
+    sparseCheckout = [
+      "sh/tools/simple-dlna-browser"
+    ];
+    rev = "666e2c402723251ba19ce9d7c11b83b06c15a2c5";
+    sha256 = "sha256-jwPAEgti5DrjNnNVW/61GGTP+QedGAjz9kPwYhMAdTk=";
+  };
+
+  buildInputs = buildInputs;
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/simple-dlna-browser/default.nix:36]: Please inline

File: pkgs/tools/networking/simple-dlna-browser/default.nix [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+  installPhase = ''
+    outputWrapper="$out/bin/simple-dlna-browser"
+    outputWrapped="$out/bin/wrapped-simple-dlna-browser"
+
+    # Use a wrapper script to prefix PATH with the buildInput
+    # This is easier than trying to modify the script with sed
+    mkdir -p $out/bin
+    echo '#!/usr/bin/env bash
+    export PATH="${paths}:''${PATH:-}"' > $outputWrapper
+    echo exec $out/bin/wrapped-simple-dlna-browser '"''${@}"' >> $outputWrapper
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/simple-dlna-browser/default.nix:47]: Use makeWrapper

File: pkgs/tools/networking/simple-dlna-browser/default.nix [hunk 7]
```diff
@@ -0,0 +1,57 @@
+{ pkgs, lib, ...}:
+let
+  buildInputs = with pkgs; [
+    gnused
+    socat
+  ];
+  binPaths = builtins.map (x: "${x}/bin") buildInputs;
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/simple-dlna-browser/default.nix:7]: Use lib.makeBInPath

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please follow the [contributing guide](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes) when naming your commits.

File: pkgs/tools/networking/simple-dlna-browser/default.nix (hunk 3 above)
Comment (LoveIsGrief) [reply pkgs/tools/networking/simple-dlna-browser/default.nix:11]: There is no version. Should I take the commit ID as the version?

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (LoveIsGrief) [reply maintainers/maintainer-list.nix:8349]: Done in 7df27f1

File: pkgs/tools/networking/simple-dlna-browser/default.nix (hunk 7 above)
Comment (LoveIsGrief) [reply pkgs/tools/networking/simple-dlna-browser/default.nix]: Done in 9ca8ba8

File: pkgs/tools/networking/simple-dlna-browser/default.nix (hunk 6 above)
Comment (LoveIsGrief) [reply pkgs/tools/networking/simple-dlna-browser/default.nix]: Done in 9ca8ba8

File: pkgs/tools/networking/simple-dlna-browser/default.nix (hunk 5 above)
Comment (LoveIsGrief) [reply pkgs/tools/networking/simple-dlna-browser/default.nix]: Done in 9ca8ba8

File: pkgs/tools/networking/simple-dlna-browser/default.nix (hunk 4 above)
Comment (LoveIsGrief) [reply pkgs/tools/networking/simple-dlna-browser/default.nix]: Done in 9ca8ba8

File: pkgs/tools/networking/simple-dlna-browser/default.nix (hunk 2 above)
Comment (LoveIsGrief) [reply pkgs/tools/networking/simple-dlna-browser/default.nix]: Done in 9ca8ba8

File: pkgs/tools/networking/simple-dlna-browser/default.nix (hunk 3 above)
Comment (SuperSandro2000) [reply pkgs/tools/networking/simple-dlna-browser/default.nix:9]: Please take the date of the commit like https://github.com/NixOS/nixpkgs/blob/ee96940bad52677b362baad3150a0931abb74350/pkgs/applications/audio/adlplug/default.nix#L49

Comment (LoveIsGrief) [reply pkgs/tools/networking/simple-dlna-browser/default.nix]: Done in 
0f356ab

**Comment by LoveIsGrief** [comment]
Thanks for the review @SuperSandro2000 !

File: pkgs/tools/networking/simple-dlna-browser/default.nix [hunk 8]
```diff
@@ -0,0 +1,54 @@
+{ pkgs, lib, makeWrapper, ...}:
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/simple-dlna-browser/default.nix:1]: pkgs should not be used as an input in normal packages. Also ... is not necessary here.

File: pkgs/tools/networking/simple-dlna-browser/default.nix [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+    owner = "javier-lopez";
+    repo = "learn";
+    sparseCheckout = [
+      "sh/tools/simple-dlna-browser"
+    ];
+    rev = "666e2c402723251ba19ce9d7c11b83b06c15a2c5";
+    sha256 = "sha256-jwPAEgti5DrjNnNVW/61GGTP+QedGAjz9kPwYhMAdTk=";
+  };
+
+  inherit (buildInputs);
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/simple-dlna-browser/default.nix:22]: Are those packages used at build time?

File: pkgs/tools/networking/simple-dlna-browser/default.nix [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+  nativeBuildInputs = [ makeWrapper ];
+
+  installPhase = ''
+    outputWrapper="$out/bin/simple-dlna-browser"
+    outputWrapped="$out/bin/wrapped-simple-dlna-browser"
+
+    mkdir -p $out/bin
+    cp sh/tools/simple-dlna-browser $outputWrapped
+    makeWrapper $outputWrapped $outputWrapper \
+      --prefix PATH : ${lib.makeBinPath buildInputs}
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/simple-dlna-browser/default.nix:32]: If they are only used for wrapping here then buildInputs above is doing nothing and we can simplify this to 

```suggestion
      --prefix PATH : ${lib.makeBinPath [ gnused socat ]}
```

File: pkgs/tools/networking/simple-dlna-browser/default.nix [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+
+  meta = {
+    description = "Query dlna media servers";
+    longDescription = ''
+      A bash script that allows simple interactions with [DLNA] media servers.
+      It is a light-weight, CLI alternative to [VLC], [Kodi], and other players.
+
+      [DLNA]: https://en.wikipedia.org/wiki/Digital_Living_Network_Alliance
+      [Kodi]: https://kodi.tv/
+      [VLC]: https://www.videolan.org/vlc/
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/simple-dlna-browser/default.nix:47]: I am not sure if the longDescription is actual markdown. It could also be that it is just written as is into the result.

**Review by SuperSandro2000 — COMMENTED** [review]
I missed some things the first time, sorry.

File: pkgs/tools/networking/simple-dlna-browser/default.nix (hunk 11 above)
Comment (LoveIsGrief) [reply pkgs/tools/networking/simple-dlna-browser/default.nix:47]: The [documentation](https://nixos.org/manual/nixpkgs/stable/#var-meta-longDescription) says it's CommonMark.

> An arbitrarily long description of the package in [CommonMark](https://commonmark.org/) Markdown.

File: pkgs/tools/networking/simple-dlna-browser/default.nix (hunk 10 above)
Comment (LoveIsGrief) [reply pkgs/tools/networking/simple-dlna-browser/default.nix:32]: I just tried this, but got `error: undefined variable 'gnused'`. How does one access these packages without using `pkgs` ?

File: pkgs/tools/networking/simple-dlna-browser/default.nix (hunk 9 above)
Comment (LoveIsGrief) [reply pkgs/tools/networking/simple-dlna-browser/default.nix:22]: `sed` is used in the install phase and `socat` is necessary for the script's full functionality.

File: pkgs/tools/networking/simple-dlna-browser/default.nix (hunk 10 above)
Comment (LoveIsGrief) [reply pkgs/tools/networking/simple-dlna-browser/default.nix]: Ah... I get it now. `callPackage` uses dependency injection. We desperately need documentation on the pattern as well as to generate documentation from nix code like `javadoc` and `cargo doc`...

Anyway, this should work now in 268feff

File: pkgs/tools/networking/simple-dlna-browser/default.nix (hunk 9 above)
Comment (LoveIsGrief) [reply pkgs/tools/networking/simple-dlna-browser/default.nix]: Found a few more dependencies in the script and added them to the args. Nothing is really built, so I could remove the line entirely in 268feff

**Comment by LoveIsGrief** [comment]
That's alright, I learned something new. Thanks again
